Tutorial (#293)
cyclus/cyclus.github.com: source/user/tutorial/sim_parm.rst

> * mine of natural uranium

Reviewer: ```suggestion
* mine producing natural uranium
```

More details about each of these facilities will discussed when we are
required to provide that input.

Concept: Simulation Time Steps
------------------------------

> CYCLUS uses a time-step approach to march through time and determine what

Reviewer 2: ```suggestion
|Cyclus| uses a time-step approach to march through time and determine what
```

2. <duration>: the number of months to be simulated

3. <startmonth>: the first month of the simulation (e.g.: 1 for January)

4. <startyear>: the first year of the simulation

5. <decay>:The Cyclus kernel has built-in experimental support for `Decay <http://fuelcycle.org/devdoc/decay.html>`_ calculations. Materials store the time since their last decay and agents are free to invoke the decay function on them as desired to decay them to the current simulation time. Cyclus can operate in 3 decay modes, with 1 additional mode likely to be added in a future release:

- 'never' if all decay is turned off
- 'manual', meaning it is only on if the individual archetype decays their own inventory
- 'lazy', which will compute decay only when archetypes fetch a particular composition.
- 'periodic' (future) automatically decays all materials in a simulation with some fixed frequency. 

There are other `optional parameters <http://fuelcycle.org/user/input_specs/control.html>`_ that
could be given but are not in the scope of this tutorial.

6. Description: A brief description of your simulation.

We'll return later to the topics of generating, loading and executing an input file.

> The simulation control is the first part of the CYCLUS input file and is of the form:

Reviewer 2: ```suggestion
The Cyclus input file begins with the ``<simulation>`` tag and ends with the ``</simulation>`` tag.
Within this space, the ``<control>`` block is the first section of the CYCLUS input file and is of the form:
```

> ::

Reviewer 2: ```suggestion
.. code-block:: XML
```

    <simulation>
      <control>
        <duration>duration_val</duration>
        <startmonth>start_month_val</startmonth>
        <startyear>start_year_val</startyear>
        <decay>decay_val</decay>
      </control>

> 

Reviewer 2: ```suggestion
```

    </simulation>

The lifetime of a *CYCLUS* simulation is determined by its

> **duration**. **duration** is the number of months *CYCLUS* will

Reviewer 2: ```suggestion
**duration**, the number of months *CYCLUS* will
```

model the fuel cycle. *CYCLUS* also intakes the **start_month** and
**start_year** of the simulation. The last major parameter of the
simulation is whether or not we wish to model the
`decay <http://fuelcycle.org/devdoc/decay.html>`__ of the
radioactive elements (uranium ore, nuclear fuel, & spent nuclear fuel)
in the simulation. For simplicity, we will not model decay in this
tutorial.

Activity: Set Simulation Parameters
-----------------------------------

> Using the simulation control template above and the table below, properly fill the template with the variables listed in the table below in your favorite text editor

Reviewer 2: ```suggestion
Using the simulation control template above and the table below, properly fill the template 
with the variables listed in the table below in your favorite text editor.
```

+-------------------+---------------+---------------------------------+
| Variable          | Value         | Purpose                         |
+===================+===============+=================================+
| ``duration``      | ``720``       | length of simulation (months)   |
+-------------------+---------------+---------------------------------+
| ``start_month``   | ``1``         | start month of simulation       |
+-------------------+---------------+---------------------------------+
| ``start_year``    | ``2018``      | start year of simulation        |
+-------------------+---------------+---------------------------------+
| ``decay``         | ``never``     | radioactive decay               |
+-------------------+---------------+---------------------------------+

Using this table, let's set the simulation parameters.

1. To tell CYCLUS that this is the simulation section of the input file, set  the first line of the input file to be:

> ::

Reviewer 2: ```suggestion
.. code-block:: XML
```

    <simulation>

2. Place the ``control`` header in as such

> ::

Reviewer 2: ```suggestion
.. code-block:: XML
```

  <simulation>
    <control>

> If you would like, you can space or tab the ``control`` header over for personal ease of reading the file but it is not necessary.

Reviewer 2: ```suggestion
Adding spaces or tabs to indent the ``control`` header improves ease of reading. 
```

> 3. After filling in the parameters listed in the table above tab out and close the control and simulation sections as:

Reviewer 2: ```suggestion
3. After filling in the parameters listed in the table above, close the control and simulation sections as:
```

> ::

Reviewer 2: ```suggestion
.. code-block:: XML
```

    <simulation>
      <control>
        <duration>720</duration>
        <startmonth>1</startmonth>
        <startyear>2018</startyear>
        <decay>never</decay>
      </control>

    </simulation>

> **Note**: There is two spaces between the end of the control section and end of the simulation section as the simulation section will hold the commodities, facilities, regions, institutions, and recipe information.

Reviewer 2: ```suggestion
**Note**: There are two blank lines between the end of the control section and end of the simulation section. This ection of the simulation block will hold the rest of the simulation parameter blocks (commodities, facilities, regions, institutions, and recipe blocks).
```

| ``start_month``   | ``1``         | start month of simulation       |
+-------------------+---------------+---------------------------------+
| ``start_year``    | ``2018``      | start year of simulation        |
+-------------------+---------------+---------------------------------+
| ``decay``         | ``never``     | radioactive decay               |
+-------------------+---------------+---------------------------------+

Using this table, let's set the simulation parameters.

1. To tell CYCLUS that this is the simulation section of the input file, set  the first line of the input file to be:
.. code-block:: XML

    <simulation>

2. Place the ``control`` header in as such

.. code-block:: XML

  <simulation>
    <control>

Adding spaces or tabs to indent the ``control`` header improves ease of reading. 

3. After filling in the parameters listed in the table above, close the control and simulation sections as:

.. code-block:: XML

    <simulation>
      <control>
        <duration>720</duration>
        <startmonth>1</startmonth>
        <startyear>2018</startyear>
        <decay>never</decay>
      </control>

    </simulation>

> **Note**: There are two blank lines between the end of the control section and end of the simulation section. This ection of the simulation block will hold the rest of the simulation parameter blocks (commodities, facilities, regions, institutions, and recipe blocks).

Reviewer 3: ```suggestion
**Note**: There are two blank lines between the end of the control section and end of the simulation section. This section of the simulation block will hold the rest of the simulation parameter blocks (commodities, facilities, regions, institutions, and recipe blocks).
```

Users do not have to manage these phases, but must provide the following
information for all simulations:

1. <simulation>:  The simulation handle contains all the parameters in the simulation.

2. <duration>: the number of months to be simulated

3. <startmonth>: the first month of the simulation (e.g.: 1 for January)

4. <startyear>: the first year of the simulation

> 5. <decay>:The |Cyclus| kernel has built-in experimental support for `Decay <http://fuelcycle.org/devdoc/decay.html>`_ calculations. Materials store the time since their last decay and agents are free to invoke the decay function on them as desired to decay them to the current simulation time. Cyclus can operate in 3 decay modes, with 1 additional mode likely to be added in a future release:

Reviewer: ```suggestion
5. <decay>:The |Cyclus| kernel has built-in experimental support for `Decay <http://fuelcycle.org/devdoc/decay.html>`_ calculations. Materials store the time since their last decay and agents are free to invoke the decay function on them as desired to decay them to the current simulation time. |Cyclus| can operate in 3 decay modes, with 1 additional mode likely to be added in a future release:
```

> - 'never' if all decay is turned off

Reviewer: ```suggestion
- 'never', all decay is turned off
```

- 'manual', meaning it is only on if the individual archetype decays their own inventory
- 'lazy', which will compute decay only when archetypes fetch a particular composition.

> - 'periodic' (future) automatically decays all materials in a simulation with some fixed frequency. 

Reviewer: ```suggestion
- 'periodic' (future), automatically decays all materials in a simulation with some fixed frequency. 
```

There are other `optional parameters <http://fuelcycle.org/user/input_specs/control.html>`_ that
could be given but are not in the scope of this tutorial.

6. Description: A brief description of your simulation.

We'll return later to the topics of generating, loading and executing an input file.

A |Cyclus| input file can be written in `multiple formats <https://fuelcycle.org/user/writing_input.rst>`_, 

> including XML, JSON, and python. This tutorial

Reviewer: ```suggestion
including XML, JSON, and Python. This tutorial
```

Users do not have to manage these phases, but must provide the following
information for all simulations:

1. <simulation>:  The simulation handle contains all the parameters in the simulation.

> 2. <duration>: the number of months to be simulated

Reviewer: OO I just wondered about that today as I was giving a live demo to some LANL folks.... it also says "duration (required once) - the time duration of the simulation (in months)" in the [user guide page on control](http://fuelcycle.org/user/input_specs/control.html)

3. <startmonth>: the first month of the simulation (e.g.: 1 for January)

4. <startyear>: the first year of the simulation

5. <decay>:The |Cyclus| kernel has built-in experimental support for 
`Decay <http://fuelcycle.org/devdoc/decay.html>`_ calculations. Materials 
store the time since their last decay and agents are free to invoke the 
decay function on them as desired to decay them to the current simulation 
time. |Cyclus| can operate in 3 decay modes, with 1 additional mode 
likely to be added in a future release:

- 'never', all decay is turned off
- 'manual', meaning it is only on if the individual archetype decays their own inventory
- 'lazy', which will compute decay only when archetypes fetch a particular composition.
- 'periodic' (future), automatically decays all materials in a simulation with some fixed frequency. 

There are other `optional parameters <http://fuelcycle.org/user/input_specs/control.html>`_ that
could be given but are not in the scope of this tutorial.

> 6. Description: A brief description of your simulation.

Reviewer 4: I'm not sure. I think this is a relic from the description field of cyclist. I don't think I have seen an xml input with this field. Does it still exist? Does anyone use it? Should I just delete this?

> 6. Description: A brief description of your simulation.

Reviewer 3: I think we should just delete this
